controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -225,7 +225,7 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 			_, scIsFSType := existing.Parameters["fsName"]
 			scEncryptionMethod, scHasEncryptionMethod := existing.Parameters["encryptionMethod"]
 			if !((sccType == "sharedfilesystem" && scIsFSType && !scHasEncryptionMethod) ||
-				(sccType == "blockpool" && !scIsFSType && sccEncryptionMethod != scEncryptionMethod)) {
+				(sccType == "blockpool" && !scIsFSType && sccEncryptionMethod == scEncryptionMethod)) {
```

Comment:
I and @rexagod discuss this issue(which I faced during testing) and made this change. 
Also tested with above change everything works before this change storageClassClaim creation was failing
for rbd in case of upgrade from `4.10` to `4.11`.

Reply:
cc @Rakshith-R

Reply:
I'd prefer we rework the entire if statement condition, It is definitely not easy to read.

@subhamkrai 
Have you checked why upgrade is failing ? 
Paste the yamls of both claim and sc before upgrade and expectation after upgrade.

@rexagod 
Any reason to think why it is ok to revert the change made earlier ?

Reply:
@subhamkrai Can you try creating a encryption enabled claim with this check ?

Reply:
PLMK if you think that's not the case.

Reply:
But does it succeed in creating a encrypted sc now ? 

If it does, then its cool.

Reply:
@rexagod @Rakshith-R made the changes as discussed with @nb-ohad  should make more sense now

Reply:
@subhamkrai @rexagod 
Please validate both creation of encrypted and non-encrypted storageclassclaims.

@rexagod Then this pr may need to be visited again ? 
https://github.com/red-hat-storage/ocs-operator/pull/1727

Reply:
Yes, we will most probably need to backport this change after verification, if this is indeed the fix.